### ISSUE TITLE
DateInput. Не сохраняется выбранный год/месяц в режиме инпута "месяц.год" при попытке изменить только год или только месяц

### ISSUE THREAD

**Issue by gh-igor** [AdmiralDS/react-ui#1286]
В [сторибуке](https://admiralds.github.io/react-ui/?path=/story/admiral-2-1-input-dateinput--date-input-pick-month) описан пример инпута в режиме ввода месяца и года.

Выявлены 2 неудобства:

1. При модификации даты с 03.2025 на 04.2025 ожидается, что достаточно выбрать месяц в два клика:
![image](https://github.com/AdmiralDS/react-ui/assets/74675573/c4a39978-2786-444b-8073-a9bb3a8d64a8)

    Однако в таком случае год не сохраняется и подставляется текущий (2024), при том что в диалоговом окне выбора года выделен 2025-й:
![image](https://github.com/AdmiralDS/react-ui/assets/74675573/d3ba0450-4682-4234-be3c-3000cc3a3c94)

2. При модификации даты с 03.2025 на 03.2026 ожидается, что достаточно выбрать год модальном окне года, однако в довесок к этому происходит сброс месяца до января, при том что в модальном окне выбора месяца выделен март:
![image](https://github.com/AdmiralDS/react-ui/assets/74675573/eca60fa1-7b1a-416d-b351-b6737ae74a0f)

**Comment by DrUNE** [comment]
Да, есть проблема, это по причине не достаточного гибкого апи, сечас идет переработка календаря и дейтпикера снуля

**Comment by DrUNE** [comment]
Будет реализовано в рамках https://github.com/AdmiralDS/date-picker/issues/21
